### PR TITLE
fix(mechanics): Properly reset player flagship after death

### PR DESCRIPTION
**Pull request by tibetiroka**
**Bug fix**

This PR addresses the bug described in issue #10125 

## Summary
Resetting the player flagship seems to fix most issues, except:
 - the ammo counter is still visible (but cannot be interacted with)
 - the trading panel is still available

## Testing Done
Tested with a mission that kills the player on a planet. Needs testing with mid-flight death.

### CODE REVIEW

**Comment by TomGoodIdea** [comment]
You can't depart wihtout a flagship, but you still have access to some other options, like the trading panel
Screenshot_20240530_210444

**Comment by TomGoodIdea** [comment]
> The other two may be done in a separate PR, maybe we just drop the planet panel on death

I think destroying the planet panel on death would fix this issue as well, because without a planet panel the player wouldn't be able to depart

**Comment by TomGoodIdea** [comment]
Simple flagship reset isn't solving anything, here's a screenshot from the outfitter:

And here's your save from #10125, but moved the player to Luna:
[Matthew_WeaverGhost.txt](https://github.com/user-attachments/files/15506720/Matthew_WeaverGhost.txt)
There's a shipyard on Luna, so you can just go there, buy a new ship and depart, but you're still considered dead.

**Comment by tibetiroka** [comment]
> You can't depart wihtout a flagship, but you still have access to some other options, like the trading panel

I can't reproduce that. When I die on a planet, I don't see the trading panel, neither in this PR on on master.

**Comment by tibetiroka** [comment]
I think I got all the necessary UI fixes in as well. Do you have a save file for a mid-flight death for testing?

**Comment by RisingLeaf** [comment]
I would still prefer if you popped the planet panel instead of blocking input

**Comment by tibetiroka** [comment]
Looks like I need to do both.

File: source/PlanetPanel.cpp [hunk 1]
```diff
@@ -103,7 +103,10 @@ void PlanetPanel::Step()
 void PlanetPanel::Draw()
 {
 	if(player.IsDead())
+	{
+		GetUI()->Pop(this);
```
Comment (RisingLeaf) [inline source/PlanetPanel.cpp:107]: This should be a popThrough and inside step

File: source/MainPanel.cpp [hunk 2]
```diff
@@ -71,6 +71,8 @@ void MainPanel::Step()
 	// checks only already-drawn panels.
 	bool isActive = GetUI()->IsTop(this);
 
+	if(player.IsDead())
+		show = Command::NONE;
```
Comment (RisingLeaf) [inline source/MainPanel.cpp:75]:
```suggestion
		show = Command::NONE;

```
And a comment pls

File: source/MainPanel.cpp [hunk 3]
```diff
@@ -203,6 +205,8 @@ Engine &MainPanel::GetEngine()
 // Only override the ones you need; the default action is to return false.
 bool MainPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &command, bool isNewPress)
 {
+	if(player.IsDead())
+		return true;
```
Comment (RisingLeaf) [inline source/MainPanel.cpp:209]:
```suggestion
		return true;

```

File: source/PlayerInfo.cpp [hunk 4]
```diff
@@ -646,6 +646,7 @@ void PlayerInfo::Die(int response, const shared_ptr<Ship> &capturer)
 		if(it != ships.end())
 			ships.erase(it);
 	}
+	flagship.reset();
```
Comment (RisingLeaf) [inline source/PlayerInfo.cpp:649]:
```suggestion

	flagship.reset();
```

**Review by RisingLeaf — CHANGES_REQUESTED** [review]

**Review by RisingLeaf — APPROVED** [review]
I am not able to test, but looks good

**Comment by tibetiroka** [comment]
Tested and working with mid-flight death as well.

File: source/MainPanel.cpp [hunk 5]
```diff
@@ -71,6 +71,9 @@ void MainPanel::Step()
 	// checks only already-drawn panels.
 	bool isActive = GetUI()->IsTop(this);
 
+	if(player.IsDead())
```
Comment (tibetiroka) [inline source/MainPanel.cpp:74]:
```suggestion
	// If the player is dead, don't show anything.
	if(player.IsDead())
```

File: source/PlanetPanel.cpp [hunk 6]
```diff
@@ -75,6 +75,13 @@ PlanetPanel::PlanetPanel(PlayerInfo &player, function<void()> callback)
 
 void PlanetPanel::Step()
 {
+	if(player.IsDead())
```
Comment (tibetiroka) [inline source/PlanetPanel.cpp:78]:
```suggestion
	// If the player is dead, pop the planet panel.
	if(player.IsDead())
```

**Review by TomGoodIdea — APPROVED** [review]
Looks like it finally works

File: source/AmmoDisplay.cpp [hunk 7]
```diff
@@ -38,10 +38,17 @@ AmmoDisplay::AmmoDisplay(PlayerInfo &player)
 }
 
 
+void AmmoDisplay::Reset()
+{
+	ammo.clear();
+}
+
+
+
```
Comment (Amazinite) [inline source/AmmoDisplay.cpp:47]:
```suggestion
```

File: source/AmmoDisplay.cpp [hunk 8]
```diff
@@ -38,10 +38,16 @@ AmmoDisplay::AmmoDisplay(PlayerInfo &player)
 }
```
Comment (Amazinite) [inline source/AmmoDisplay.cpp:40]:
```suggestion


```

**Pull request merged**